### DISCUSSION TITLE
Is it possible to manually force an action workflow to be re-run?

### DISCUSSION THREAD

**Discussion by cs-cordero** [community/community#25726] [Actions]
<p>I pushed changes to my codebase, triggering a workflow to start.  It fails.  I’d like to have it try again without pushing new code.</p>
<p>I am not expecting any difference in behavior (I’d expect it to fail again), but I want to know if manual runs are possible.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>No, re-running a GitHub Action is not currently supported. I’ll pass along the request, but can’t give an ETA on when they might implement such a feature.</p>
<p>I hope that helps!</p>

**Reply by tsal** [reply]
It's now 2023; can we PLEASE get this supported? The re-run button isn't enough, and this is the ONLY CI platform that doesn't allow manual triggering.

**Reply by case** [reply]
https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow

https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/

**Reply by jcam** [reply]
neither of these really support the use case. "oh you want to re-run this action from last month? okay, go back in time and update your workflows to support manual dispatch, then revert the primary branch to the branch/code/commit you want deployed. easy peasy!"

My current solution for a rollback is to manually stash anything recent and revert the branch to the commit i needed, even if it's a year or more old, then put the latest code back and quickly jump in and cancel the action for the updated code. it's... not a great experience...

**Comment by thasmo** [comment]
<p>Manually rerunning an action is really needed. In addition I also would like to manually run a specific workflow; i.e. I would like to test a scheduled workflow before its scheduled run.</p>

**Comment by case** [comment]
<p>+1 for this. Sometimes there are errors (e.g. due to temporary upstream outages) when things run on a schedule, and being able to just click a button to re-run would be ideal.</p>
<p>Thanks for listening!</p>

**Comment by floooh** [comment]
<p>+1, manually restarting a workflow is currently the feature I miss the most.</p>
<p>I my case it would be even better if I could start a workflow on push to *other* repositories (for instance rebuild a top-level project when dependencies have changed).</p>

**Comment by dkrantsberg** [comment]
<p>+1<br>
IMO, this is really a must feature for any CI service.</p>

**Comment by abatilo** [comment]
<p>Hi all, I’m not entirely sure if this is appropriate or not to self promote here, but my buddy and I built an extremely basic MVP for handling this exact situation, because we wanted this feature as well:</p>
<p><a href="https://www.actionspanel.app" rel="nofollow noopener">https://www.actionspanel.app</a></p>
<p>We built it as fast as we could so it’s still very quirky, but we’d love feedback. There’s a contact email on that landing page.</p>

**Comment by joaquinmoreira** [comment]
<p>Its very hard to survive without this. Please, remove the condition to show the “rerun checks” only when the workflow failed.</p>

**Comment by birnbuazn** [comment]
<p>Just fleshing out my workflow and got a lot of</p>
<pre><code>dummy commit to trigger CI/CD workflow
</code></pre>
<p>commit messages in my repo so far :smiley:</p>

**Comment by UmeshCEcho** [comment]
<p>It is very much required.</p>

**Comment by LudgerPeters** [comment]
<p>Is there any update on this feature request.</p>

**Comment by Siedlerchr** [comment]
<p>Yes, this would actualy be helpful! Simply remove the condition of failed.<br>
I found this blog post which creates a workaround for manually triggering the workflow through a curl request using the API:</p>
<p><a href="https://goobar.io/2019/12/07/manually-trigger-a-github-actions-workflow/" rel="nofollow noopener">https://goobar.io/2019/12/07/manually-trigger-a-github-actions-workflow/</a></p>

**Comment by jcam** [comment]
<p>Could you please add me and my group to this feature request?</p>
<p>Today’s github outage meant that a few of our Actions runs didn’t trigger, and now I am forced to manually delete and re-apply the commits so they trigger properly.</p>

**Comment by warpech** [comment]
<p>Solution that works for me: To re-run a successful commit, push that commit onto a new branch or add a new tag.</p>

**Comment by ch264** [comment]
<p>I second to remove the conditional re-run. In the meantime I have added a Github Action for a manual rerun that I trigger from Postman. It is a quick setup and works great. I have the steps outlined here <a href="https://medium.com/@christinavhastenrath/how-to-run-github-actions-manually-afebbe77d325" rel="nofollow noopener">https://medium.com/@christinavhastenrath/how-to-run-github-actions-manually-afebbe77d325</a> </p>

**Comment by dskvr** [comment]
<aside class="quote group-github-employees" data-username="lee-dohm" data-post="2" data-topic="2127">
<div class="title">
<div class="quote-controls"></div>
 lee-dohm:</div>
<blockquote>
<p>No, re-running a GitHub Action is not currently supported</p>
</blockquote>
</aside>
<p>Said no successful pipeline service ever. Absolutely necessary in real world use. Showstopper.  Prime example are those who could not run jobs because GitHub went down.</p>
<p>Forcing a commit is not a reasonable solution unless it’s stated that this feature is coming, and only then is it an acceptable workaround.</p>

**Comment by maknesium** [comment]
<p>+1</p>
<p>We are migrating from GitLab to GitHub and I really really miss the feature to trigger the CI/CD pipeline manually. A “re-run” button shouldn’t be too hard to implement, either :wink:</p>

**Comment by maxceem** [comment]
<p>+1</p>
<p>Trying to move from CircleCI to the Github Actions and really miss the ability to re-run the workflow instead of creating a dummy commit.</p>

**Comment by maxceem** [comment]
<p>There is a <strong>re-run</strong> button:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/0/09956a90922009ea8151a78076a3c44b9a4d28f3.png" data-download-href="/uploads/short-url/1mMq2UELkUelrY5U2yVJCFQRHfJ.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/146016/181092878-b7ef5797-2551-4797-a60d-cd71d8cd8056.png" alt="image" data-base62-sha1="1mMq2UELkUelrY5U2yVJCFQRHfJ" width="690" height="284" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/0/09956a90922009ea8151a78076a3c44b9a4d28f3_2_690x284.png, https://aws1.discourse-cdn.com/github/optimized/2X/0/09956a90922009ea8151a78076a3c44b9a4d28f3_2_1035x426.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/0/09956a90922009ea8151a78076a3c44b9a4d28f3_2_1380x568.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/0/09956a90922009ea8151a78076a3c44b9a4d28f3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">3090×1274 224 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
